### PR TITLE
Update argocd action

### PR DESCRIPTION
**Pull request by marchuk-vlad**

### CODE REVIEW

File: actions/update-argocd/action.yaml [hunk 1]
```diff
@@ -0,0 +1,163 @@
+name: Update argocd-clusters
+description: Update solution or data value in argocd cluster repo
+
+inputs:
+  gh_token:
+    description: Token for GH argo repository
+    required: true
+  target:
+    description: Specify "develop" or "stage" or "testnet"
+    required: true
```
Comment (serega-k) [inline actions/update-argocd/action.yaml:10]: @marchuk-vlad , maybe use
```yaml
        type: choice
        options:
          - stage
          - develop
          - testnet
```
will be better

Comment (marchuk-vlad) [reply actions/update-argocd/action.yaml:10]: There is no types in inputs in [composit actions](https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-composite-action). We can specify only  `description` `required` `default`.

**Review by Idris0v — APPROVED** [review]

File: actions/update-argocd/action.yaml [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      run: |
+        if [ -n "${{ inputs.cluster }}" ]; then
+          CLUSTER=${{ inputs.cluster }}
+        elif [ "${{ inputs.target }}" != "dev" ]; then
+          CLUSTER=ovh-de-dev/dev
+        elif [ "${{ inputs.target }}" != "stage" ]; then
+          CLUSTER=ovh-stg/stage
+        elif [ "${{ inputs.target }}" != "testnet" ]; then
+          CLUSTER=gcore-testnet/testnet
+        fi
```
Comment (serega-k) [inline actions/update-argocd/action.yaml]:
```suggestion
        if [ -n "${{ inputs.cluster }}" ]; then
          CLUSTER=${{ inputs.cluster }}
        elif [ "${{ inputs.target }}" == "dev" ]; then
          CLUSTER=ovh-de-dev/dev
        elif [ "${{ inputs.target }}" == "stage" ]; then
          CLUSTER=ovh-stg/stage
        elif [ "${{ inputs.target }}" == "testnet" ]; then
          CLUSTER=gcore-testnet/testnet
        fi
```

File: actions/update-argocd/action.yaml [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    - name: Modify cluster files
+      shell: bash
+      run: |
+        yq --version
+        jq --version
+
+        WORKDIR=$(pwd)
+        DST=$WORKDIR/argo/clusters/$CLUSTER
+
+        cp "$DST/execution-controller-${{ inputs.resource_type }}-prov.yml" $WORKDIR/ec.yaml
```
Comment (serega-k) [inline actions/update-argocd/action.yaml]:
```shall
FILE="$DST/execution-controller-${{ inputs.resource_type }}-prov.yml"
if [[ -f $FILE ]]; then
  cp $FILE $WORKDIR/ec.yaml
else
  echo "File $FILE does not exist"
  exit 1
fi
```

File: actions/update-argocd/action.yaml [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+        # extract helm YAML values from ArgoCD app, and extract provider offers JSON config from YAML helm config
+        yq eval '.spec.source.helm.values' ec.yaml > config.yaml
```
Comment (serega-k) [inline actions/update-argocd/action.yaml]:
```suggestion
RESULT=$(yq eval '.spec.source.helm.values' ec.yaml)

if [[ -z "$RESULT" ]]; then
  echo "The evaluation result is empty"
  exit 1
else
  echo "$RESULT" > config.yaml
fi
```

File: actions/update-argocd/action.yaml [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        yq eval '.config.PROVIDER_OFFERS_JSON' config.yaml > provider-offers-config.json
```
Comment (serega-k) [inline actions/update-argocd/action.yaml]: the same suggestion as above

File: actions/update-argocd/action.yaml [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        # get offer id by solution instance fullpath (select block, which should be replaced in JSON config)
+        offer_id=$(jq ".[] | select(.resource.filepath | contains(\"${{ inputs.solution_name }}\")) | .id" provider-offers-config.json)
+        if [ -z "$offer_id" ]; then
+          echo "Offer id is not found"
+          exit 1
+        fi
+
+        encryption_block=$(cat ${{ inputs.resource_file }} | jq ".encryption")
+        resource_block=$(cat ${{ inputs.resource_file }} | jq ".resource")
+        hash_block=$(cat ${{ inputs.resource_file }} | jq ".hash")
```
Comment (serega-k) [inline actions/update-argocd/action.yaml]:
```shell
json_content=$(cat ${{ inputs.resource_file }})
encryption_block=$(echo "$json_content" | jq ".encryption")
resource_block=$(echo "$json_content" | jq ".resource")
hash_block=$(echo "$json_content" | jq ".hash")
```

**Review by serega-k — APPROVED** [review]

**Pull request merged**